PayArc: Formatting CC month, adding tax_rate, removing default void reason

jessiagee opened the pull request:
* Fixed an issue with CC months not working for '01', '02', etc.
* Added `tax_rate` to the charges allowed list
* Removed the `duplicate` default void reason

Loaded suite test/remote/gateways/remote_pay_arc_test

`test_successful_credit` fails due to MID configuration on PayArc's side

22 tests, 54 assertions, 1 failures, 0 errors, 0 pendings, 0 omissions, 0 notifications
95.4545% passed

Loaded suite test/unit/gateways/pay_arc_test

14 tests, 42 assertions, 0 failures, 0 errors, 0 pendings, 0 omissions, 0 notifications
100% passed

bundle exec rake test:local

4835 tests, 73897 assertions, 0 failures, 0 errors, 0 pendings, 0 omissions, 0 notifications
100% passed

Running RuboCop...
Inspecting 707 files

707 files inspected, no offenses detected

jessiagee commented:
@senthil-code Hey Senthil, do you see any issues here?

meagabeth reviewed (changes requested):
These changes look good, but I think we should revise the unit test to show that the month being sent is being modified to two digits. As of now, we tell it to respond with a successful response and so therefore it succeeds, but we're not showing that the month is being adjusted.

cdm-83 commented:
> @senthil-code Hey Senthil, do you see any issues here?

@jessiagee - Looks fine, No issues.

jessiagee commented:
> These changes look good, but I think we should revise the unit test to show that the month being sent is being modified to two digits. As of now, we tell it to respond with a successful response and so therefore it succeeds, but we're not showing that the month is being adjusted.

Thanks, I moved it from unit to remote (wasn't paying attention to which one I put it in, sorry!) and also I've moved https://github.com/activemerchant/active_merchant/pull/4054 into here to consolidate these small fixes into one update.

meagabeth commented on test/remote/gateways/remote_pay_arc_test.rb:
```diff
@@ -51,6 +51,57 @@ def test_successful_purchase_with_more_options
+    end
+  end
+
+  def test_failed_three_digit_card_exp_month
```
nice!

meagabeth reviewed (approved):
Looks good! 1 failing remote test due to account setup, but that was expected per our chat.
